harness/determined/deploy/gke/cli.py

```diff
@@ -0,0 +1,596 @@
+    return pod_spec
+
+
+def validate_accelerator_type(s: str) -> None:
+    json_value = subprocess.check_output(
+        ["gcloud", "compute", "accelerator-types", "list", "--format=json(name)"]
+    )
+    json_names = json.loads(json_value)
+    valid_accelerator_types = {accelerator["name"] for accelerator in json_names}
+
+    if s not in valid_accelerator_types:
+        raise ValueError("Accelerator must be one of {}".format(valid_accelerator_types))
+
+
+def validate_location(location: str, isZone: bool = True) -> None:
+    try:
+        cmd = ["gcloud", "compute"]
+        if isZone:
+            cmd += ["zones"]
+        else:
+            cmd += ["regions"]
+        cmd += ["describe", location]
+        subprocess.check_call(cmd, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+    except subprocess.CalledProcessError:
+        raise ValueError(
+            "The specified {} {} was not found".format("zone" if isZone else "region", location)
+        )
+
+
+def validate_machine_type(machine_type: str, zone: str) -> None:
+    try:
+        subprocess.check_call(
+            [
+                "gcloud",
+                "compute",
+                "machine-types",
+                "describe",
+                machine_type,
+                "--zone",
+                zone,
+            ],
+            stdout=subprocess.DEVNULL,
+        )
+    except subprocess.CalledProcessError:
+        raise ValueError(
+            "The specified machine type {} was not found in the specified zone {}.".format(
+                machine_type, zone
+            ),
+        )
+
+
+def validate_accelerator_zone(args: argparse.Namespace, zone: str) -> None:
+    try:
+        subprocess.check_call(
+            [
+                "gcloud",
+                "compute",
+                "accelerator-types",
+                "describe",
+                args.gpu_type,
+                "--zone",
+                zone,
+            ],
+            stdout=subprocess.DEVNULL,
+        )
+    except subprocess.CalledProcessError:
+        raise ValueError(
+            "The specified accelerator type {} was not found in the specified zone {}.".format(
+                args.gpu_type, zone
+            ),
+        )
+
+
+def validate_args(args: argparse.Namespace) -> None:
+    validate_location(args.zone, isZone=True)
+    validate_accelerator_type(args.gpu_type)
+
+    if args.master_machine_type != "n1-standard-16":
+        validate_machine_type(args.master_machine_type, args.zone)
+
+    if args.agent_machine_type != "n1-standard-32":
+        validate_machine_type(args.agent_machine_type, args.zone)
+
+    validate_accelerator_zone(args, args.zone)
+
+    if args.gpu_coscheduler and args.preemption:
+        raise ValueError(
+            "--gpu-coscheduler and --preemptive-scheduler are mutually exclusive and cannot both be"
+            " specified",
+        )
+
+    if args.multi_node_pool and args.cpu_coscheduler and not args.gpu_coscheduler:
```

Comment:
IIRC if you've got 2 pools, and you want a coscheduler on CPUs, you've got to have it on GPUs as well or weird things happen.

Reply:
A little late to this party, but schedulers in k8s are global, and not per resource pool. So while it is possible to run multiple schedulers in a cluster, the current state of determined (since k8s doesn't have resource pools yet) is that we can only easily use one type of scheduler at a time.